File: smart-facility-backend/src/main/java/com/example/smartfacility/service/UserService.java
package com.example.smartfacility.service;

import com.example.smartfacility.dto.request.ChangePasswordRequest;
import com.example.smartfacility.dto.request.CreateUserRequest;
import com.example.smartfacility.dto.request.UpdateUserRequest;
import com.example.smartfacility.dto.response.UserResponse;

import java.util.List;

public interface UserService {

    UserResponse createUser(CreateUserRequest request);

    List<UserResponse> getAllUsers();

    UserResponse getUserById(Long id);

    UserResponse updateUser(Long id, UpdateUserRequest request);

    UserResponse updateUserStatus(Long id, boolean active);

    UserResponse updateUserRole(Long id, com.example.smartfacility.enums.Role role);

    void deleteUser(Long id);

    UserResponse getProfile(String email);

    UserResponse updateProfile(String email, UpdateUserRequest request);

    UserResponse updateProfileImage(String email, String imageUrl);


    void changePassword(String email, ChangePasswordRequest request);

    List<UserResponse> getAvailableTechnicians(com.example.smartfacility.enums.TechnicianSpecialization specialization);
}

File: smart-facility-frontend/src/services/userService.js
import apiClient from './apiClient';

export const userService = {
  createUser: async (payload) => {
    const res = await apiClient.post('/users', payload);
    return res.data.data;
  },

  getAllUsers: async () => {
    const res = await apiClient.get('/users');
    return res.data.data;
  },

  getUserById: async (id) => {
    const res = await apiClient.get(`/users/${id}`);
    return res.data.data;
  },

  updateUser: async (id, payload) => {
    const res = await apiClient.put(`/users/${id}`, payload);
    return res.data.data;
  },

  updateUserStatus: async (id, active) => {
    const res = await apiClient.put(`/users/${id}/status`, { active });
    return res.data.data;
  },

  updateUserRole: async (id, role) => {
    const res = await apiClient.put(`/users/${id}/role`, { role });
    return res.data.data;
  },

  deleteUser: async (id) => {
    const res = await apiClient.delete(`/users/${id}`);
    return res.data;
  },

  getAvailableTechnicians: async (specialization) => {
    const res = await apiClient.get('/users/technicians', {
      params: specialization ? { specialization } : {},
    });
    return res.data.data;
  },

  getProfile: async () => {
    const res = await apiClient.get('/users/profile');
    return res.data.data;
  },

  updateProfile: async (payload) => {
    const res = await apiClient.put('/users/profile', payload);
    return res.data.data;
  },

  changePassword: async (currentPassword, newPassword) => {
    const res = await apiClient.put('/users/profile/password', { currentPassword, newPassword });
    return res.data;
  },
};
